runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/JobStateEvent.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.fnexecution.jobsubmission;
+
+import com.google.auto.value.AutoValue;
+import java.time.Instant;
+import org.apache.beam.model.jobmanagement.v1.JobApi;
+import org.apache.beam.vendor.grpc.v1p21p0.com.google.protobuf.Timestamp;
+
+/** A state transition event. */
+@AutoValue
+public abstract class JobStateEvent {
```

Comment:
We could rename the proto GetJobStateResponse to JobStateEvent and instead use that proto instead of creating this class.

Reply:
Do we want a `GetJobStateResponse` to have a `JobStateEvent` ?  i.e.

```proto
message GetJobStateResponse {
  JobStateEvent state_event = 1; // (required)
}
```

Reply:
> use that proto instead of creating this class.

I've noticed that there is a pattern in the Java and Python SDKs to avoid using protobuf objects directly as part of the Beam SDK, and instead favor "native" types with methods for converting to/from protobuf representations.  I was trying to maintain that separation by providing this `JobStateEvent` class, but I'm happy to use a protobuf object directly.  Let me know what you think.

Reply:
> We could rename the proto GetJobStateResponse to JobStateEvent and instead use that proto instead of creating this class.

Done.